test/python/result/test_result.py

```diff
@@ -121,7 +121,8 @@ def test_result_repr(self):
             "results=[ExperimentResult(shots=14, success=True, "
             "meas_level=2, data=ExperimentResultData(counts={'0x0': 4,"
             " '0x2': 10}), header=QobjExperimentHeader(creg_sizes="
-            "[['c0', 2], ['c0', 1], ['c1', 1]], memory_slots=4))])"
+            "[['c0', 2], ['c0', 1], ['c1', 1]], memory_slots=4))], date=None, "
+            "status=None, status=None)"
```

Comment:
It looks like there are some bugs in the `__repr__` method of `Result` (`status` is written twice).  Would you mind fixing that in this PR, since we're touching the object already?  Quickly looking at the rest of the `Result` object, there's also a bunch of `hasattr` checks that can be removed now, since your PR updates everything to always have these attributes.